client/app/hosting/user-logs/hosting-user-logs.controller.js

```diff
@@ -70,32 +65,34 @@ angular.module("App").controller(
         }
 
         refreshTableUserLogs () {
-            this.loaders.userLogs = true;
             this.userLogs.ids = null;
 
             return this.Hosting.getUserLogs(this.$stateParams.productId)
-                .then((data) => {
-                    this.userLogs.ids = this.$filter("orderBy")(data);
+                .then((data) => data.sort())
```

Comment:
you should check that data is indeed an array before using the `sort` method. You should also be aware of the shortcomings of using `sort`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Isn't Lodash more appropriate ?

Reply:
I would have preferred an argument like «sorting could be done by the service», If the Api is giving me a 200 without being an Array, screw it, I don't care. 

I think  ```Cannot read property 'sort' of undefined``` would be totaly fine. What do you want me to do, throw an exception if it's non an array? I don't have to write code for that.

Reply:
Well in this case you're showing the user the error `Cannot read property 'sort' of undefined` instead of showing an error message that can be translated into his language :). But I see what you mean, I've been doing the same quite a few time.

Reply:
@FredericEspiau  out of curiosity, how do you translate "a dev screwed up" into user language?  We could say an error occured while fetching ZYX which leads the user to think he can try to reload or something, but isn't the case.

Reply:
I forget to tell that in this specific case there's no «shortcomings», since it's a list of unix user with or accents or special characters. 

On that assertion problem, since javascript is an Object Language with error, I think we must escape that 90s trend of [defensive software](https://en.wikipedia.org/wiki/Defensive_programming). I use those when that kind of writing style only when I know that the API can screw-up.